similar/main/ai.cpp

```diff
@@ -3066,6 +3075,67 @@ void do_ai_frame(const vobjptridx_t obj)
 		vm_vec_zero(gun_point);
 	}
 
+	switch (robptr->boss_flag) {
+		case 0:
+			break;
+			
+		case BOSS_D1:
+			if (aip->GOAL_STATE == AIS_FLIN)
+				aip->GOAL_STATE = AIS_FIRE;
+			if (aip->CURRENT_STATE == AIS_FLIN)
+				aip->CURRENT_STATE = AIS_FIRE;
+			dist_to_player /= 4;
```

Comment:
I see that this whole block is merely moved from somewhere else, so you might not be able to answer, but I will ask anyway: Why is `dist_to_player` divided here, then reverted after the call to `do_boss_stuff`?  Since it is a local, `do_boss_stuff` cannot see its value.